### ISSUE TITLE
Build failing for PactNet.Windows Ruby file Path being too long. 

### ISSUE THREAD

**Issue by Morshed11218** [pact-foundation/pact-net#163]
Whenever I install Nuget Package Pact.Windows and try to build my solution, I'm getting the file path too long error. I even tried to move the project to smallest path possible (C:\) it still fails with that error: 

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3021	Unable to copy file "C:\Users\daudmors\.nuget\packages\pactnet.windows\2.3.3\tools\pact-win32\lib\vendor\ruby\2.2.0\gems\pact-provider-verifier-1.15.0\lib\pact\provider_verifier\underscored_headers_monkeypatch.rb" to "bin\Debug\netcoreapp2.0\pact-win32\lib\vendor\ruby\2.2.0\gems\pact-provider-verifier-1.15.0\lib\pact\provider_verifier\underscored_headers_monkeypatch.rb". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.	Ipreo.NS.OrderService.BuildingBlocks.Monitoring.UnitTests	C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets	4460

Doesn't make sense, based on where my solution is, the path count is less then 260 characters:  "C:\t\dk-ard2rservice-BuildingBlocks\src\Testing\fdreo.NS.drderrervtce.BuildingBlocks.Testing\bin\Debug\netcoreapp2.0\pact-win32\lib\vendor\ruby\2.2.0\gems\pact-provider-verifier-1.15.0\lib\pact\provider_verifier\underscored_headers_monkeypatch.rb"

**Comment by neilcampbell** [comment]
Unfortunately there is not much we can do at this stage. See https://github.com/pact-foundation/pact-net/issues/160

**Comment by vikasillumina** [comment]
@neilcampbell @bethesque @Morshed11218 I just wanted to update on this, that I tried a lot of suggestions on increasing the windows file path from default 260 characters but none of the settings seems to have made any difference. At this point I am using my mac for my development. But I do have team members who only have windows and this continues to be a problem. While I understand that we can't do much until the refactor that @bethesque mentioned in #160 but unfortunately there is no working workaround either, not at least a straight forward one. :( Just my 2 cents on this!

**Comment by bethesque** [comment]
I'm sorry @neilcampbell, I know I asked this somewhere else, but I can't find it. What happened with the idea of making the standalone path configurable?

**Comment by neilcampbell** [comment]
It's doable, just means the user would have to supply their own version on the ruby standalone. I was waiting to see how the standalone exe idea went, but sounds like its a bit of a dead end. If we think that's the best work around, lets do it.

**Comment by neilcampbell** [comment]
@vikasillumina have you tried moving your project closer to the root of your disc, like C:/{yourproject}

**Comment by vikasillumina** [comment]
Hi @neilcampbell our source repo is a mono repo containing lot of microservices. Please note our test code lives with the dev code. So in order to keep everything in same repo and organized in a structure we have longer folder paths and we can't move the project in question any much closer to root directory. Actually its already closest to the root path and problem is seen when we refer the project in other test project where it tries to copy the assemblies and run into issue.

**Comment by bethesque** [comment]
> It's doable, just means the user would have to supply their own version on the ruby standalone.

This is how Pact Go works already, as you can't package a ruby executable inside a Go executable.

**Comment by rafsojka** [comment]
@neilcampbell @bethesque 
Was the workaround discussed here ever implemented?

I was fine for one project but then I have hit the problem when trying to adopt `pact-net` in a test project for a repository with the name longer by 6 characters...

Personally I can move the project to `C:/{myproject}` on my machine but I represent here a team.
As you may guess we use .NET day to day and it may not be that easy to introduce some rules on project locations/naming due to "ruby something" :)
Hopefully you agree that is not a good way to introduce technology in enterprise environment.

**Comment by ScottDay** [comment]
Gave it a shot on my Windows machine, version 16 of msbuild (comes with Visual Studio 2019) has long path support enabled and seems to solve this issue.

Hope it helps :)

**Comment by mefellows** [comment]
> As you may guess we use .NET day to day and it may not be that easy to introduce some rules on project locations/naming due to "ruby something" :)

I understand and agree. But also, wtf Windows. Seriously. I don't get why this can't be made to be easily fixed and the new default. Let's be clear here - this is a Windows issue, and as library authors we are creating all sorts of workarounds. 

In any case, for those coming here and looking for a solution in the interim, it does appear there are options to increase the limit in certain scenarios (which I do not fully understand):

See 

* https://github.com/pact-foundation/pact-node/blob/master/README.md#enable-long-paths 
* https://github.com/pact-foundation/pact-node/issues/100 and 
* https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#maximum-path-length-limitation for more.

**Comment by fsejasm** [comment]
> @vikasillumina have you tried moving your project closer to the root of your disc, like C:/{yourproject}

It works by ny side!. Thanks!

**Comment by vikasillumina** [comment]
> > @vikasillumina have you tried moving your project closer to the root of your disc, like C:/{yourproject}
> 
> It works by my side!. Thanks!

Yes we had done it. That was one of the first suggestions and following was my response:
our source repo is a mono repo containing lot of microservices. Please note our test code lives with the dev code. So in order to keep everything in same repo and organized in a structure we have longer folder paths and we can't move the project in question any much closer to root directory. Actually its already closest to the root path and problem is seen when we refer the project in other test project where it tries to copy the assemblies and run into issue.

Unfortunately we are not using contract tests as heavily we had planned and most of us now use macbook which doesn't have the same constraint, so this issue is no longer an issue for us.

**Comment by HugoDoyon** [comment]
I proposed a fix in the this [pull request](https://github.com/pact-foundation/pact-net/pull/277). Please merge at your convenience :)
